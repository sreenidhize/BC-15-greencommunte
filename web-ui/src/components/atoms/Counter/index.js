import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Typography } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  jobs: {
    width: 200,
    height: 72,
    fontSize: 100,
    fontWeight: 300,
  },
  city: {
    width: 148,
    height: 62,
    paddingLeft: 18,
    fontWeight: 500,
    marginLeft: 10,
  },
}));

const Counter = (props) => {
  const classes = useStyles();
  const { totalJobs, city, dataTestId } = props;
  return (
    <div style={{ justifyContent: "center" }} data-testid={dataTestId}>
      {totalJobs && (
        <div>
          <Typography variant="h3" className={classes.jobs}>
            {totalJobs}
          </Typography>
        </div>
      )}
      {city && (
        <div>
          <Typography variant="h5" className={classes.city}>
            {city}
          </Typography>
        </div>
      )}
    </div>
  );
};

export default Counter;
