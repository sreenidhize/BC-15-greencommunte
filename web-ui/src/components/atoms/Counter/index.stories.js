import { ThemeProvider } from "@material-ui/core";
import React from "react";
import Counter from ".";
import Theme from "../../../utils/theme/theme";

export default {
  title: "atoms/Counter",
  component: Counter,
};

const Template = (args) => (
  <ThemeProvider theme={Theme}>
    {" "}
    <Counter {...args} />
  </ThemeProvider>
);

export const Primary = Template.bind({});
Primary.args = {
  totalJobs: 897,
  city: "Hyderabad",
};
