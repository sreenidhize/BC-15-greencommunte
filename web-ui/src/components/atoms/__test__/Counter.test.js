import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Counter from "../Counter";

const testId = "counter-placeholder";
const totalJobs = 897;
const city = "Hyderabad";

describe("Counter placeholder test", () => {
  it("Matches the snapshot", () => {
    const { container } = render(
      <Counter city={city} totalJobs={totalJobs} dataTestId={testId} />
    );
    expect(container).toMatchSnapshot();
  });

  test("test counter render", () => {
    const { getByTestId } = render(<Counter city={city} totalJobs={totalJobs} dataTestId={testId} />);
    expect(getByTestId("counter-placeholder")).toBeInTheDocument();
  });
});