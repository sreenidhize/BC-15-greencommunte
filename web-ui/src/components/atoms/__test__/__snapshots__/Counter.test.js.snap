// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter placeholder test Matches the snapshot 1`] = `
<div>
  <div
    data-testid="counter-placeholder"
    style="justify-content: center;"
  >
    <div>
      <h3
        class="MuiTypography-root makeStyles-jobs-1 MuiTypography-h3"
      >
        897
      </h3>
    </div>
    <div>
      <h5
        class="MuiTypography-root makeStyles-city-2 MuiTypography-h5"
      >
        Hyderabad
      </h5>
    </div>
  </div>
</div>
`;
